Sources/libhostmgr/CLI/Console.swift

```diff
@@ -0,0 +1,199 @@
+        terminal.clear(lines: 1)
+
+        let elapsedTime = Date().timeIntervalSince(startDate)
+
+        let rate = progress.dataRate(timeIntervalSinceStart: elapsedTime)
+        let remaining = progress.estimatedTimeRemaining(timeIntervalSinceStart: elapsedTime)
+
+        terminal.print("\(progress.formattedPercentage)% [\(rate)/s, \(Format.time(remaining))]")
+    }
+}
+
+// MARK: Static Helpers
+extension Console {
+    public static func startProgress(_ string: String) -> ProgressBar {
+        ProgressBar(title: string)
+    }
+
+    public static func startImageDownload(_ image: RemoteVMImage) -> ProgressBar {
+        let size = Format.fileBytes(image.imageObject.size)
+        return ProgressBar(title: "Downloading \(image.fileName) (\(size))")
+    }
+
+    public static func startFileDownload(_ file: S3Object) -> ProgressBar {
+        let size = Format.fileBytes(file.size)
+        return ProgressBar(title: "Downloading \(file.key) (\(size))")
+    }
+}
+
+// MARK: Static Initializers
+extension Console {
+    @discardableResult public static func heading(_ message: String) -> Self {
+        return Console().heading(message)
+    }
+
+    @discardableResult public static func success(_ message: String) -> Self {
+        return Console().success(message)
+    }
+
+    @discardableResult public static func error(_ message: String) -> Self {
+        return Console().error(message)
+    }
+
+    @discardableResult public static func warn(_ message: String) -> Self {
+        return Console().warn(message)
+    }
+
+    @discardableResult public static func info(_ message: String) -> Self {
+        return Console().info(message)
+    }
+
+    @discardableResult public static func log(_ message: String) -> Self {
+        return Console().log(message)
+    }
+
+    @discardableResult public static func printList(_ list: [String], title: String) -> Self {
+        return Console().printList(list, title: title)
+    }
+
+    @discardableResult public static func printTable(data: Table, columnTitles: [String] = []) -> Self {
+        return Console().printTable(data: data, columnTitles: columnTitles)
+    }
+
+    public static func crash(message: String, reason error: ExitCode) -> Never {
+        Console().error(message)
+        Foundation.exit(error.rawValue)
+    }
+
+    public static func exit(message: String = "", style: ConsoleStyle = .plain) -> Never {
+        Console().message(message, style: style)
+        Foundation.exit(0)
+    }
+}
+
+// MARK: Table Support
+extension Console {
+    public typealias Table = [[String]]
+
+    func columnCounts(for table: Table) -> [Int] {
+        transpose(matrix: table).map { $0.map(\.count).max() ?? 0 }
+    }
+
+    func transpose(matrix: Table) -> Table {
+        guard let numberOfColumns = matrix.first?.count else {
+            return matrix
+        }
+
+        var newTable = [[String]](repeating: [String](repeating: "", count: matrix.count), count: numberOfColumns)
```

Comment:
It took me a while to figure out what's going on here 😿 . What do you think if it's changed to
```suggestion
        var newTable = Array(repeating: Array(repeating: "", count: matrix.count), count: numberOfColumns)
```

Reply:
@crazytonyli – WDYT about 05262feba862083d2f425c813158f3e9d69df7c7?